### PR TITLE
Add cross-encoder models and refactor pretrained model section

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Add cross-encoder models and refactor pretrained model section

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _ml-commons-plugin/pretrained-models.md [hunk 1]
```diff
@@ -12,17 +12,72 @@ nav_order: 120
...
+| `ms-marco-MiniLM-L-12-v2` | 1.0.2 | - [model_url](https://artifacts.opensearch.org/models/ml-models/huggingface/cross-encoders/ms-marco-MiniLM-L-12-v2/1.0.2/torch_script/cross-encoders_ms-marco-MiniLM-L-12-v2-1.0.2-torch_script.zip) <br>- [config_url](https://artifacts.opensearch.org/models/ml-models/huggingface/cross-encoders/ms-marco-MiniLM-L-12-v2/1.0.2/torch_script/config.json) | - [model_url](https://artifacts.opensearch.org/models/ml-models/huggingface/cross-encoders/ms-marco-MiniLM-L-12-v2/1.0.2/onnx/cross-encoders_ms-marco-MiniLM-L-12-v2-1.0.2-onnx.zip) <br>- [config_url](https://artifacts.opensearch.org/models/ml-models/huggingface/cross-encoders/ms-marco-MiniLM-L-12-v2/1.0.2/onnx/config.json)
```
Comment (HenryL27) [inline _ml-commons-plugin/pretrained-models.md:74]: do we need an auto-truncation column here?

File: _ml-commons-plugin/custom-local-models.md [hunk 2]
```diff
@@ -321,148 +321,3 @@ The response contains the tokens and weights:
...
-    "model_content_hash_value": "90e39a926101d1a4e542aade0794319404689b12acfd5d7e65c03d91c668b5cf",
-    "model_config": { 
-        "model_type": "bert",
-        "embedding_dimension": 1,
-        "framework_type": "huggingface_transformers",
-        "total_chunks":2,
-        "all_config": "{\"total_chunks\":2}"
-    },
-    "url": "https://github.com/opensearch-project/ml-commons/blob/main/ml-algorithms/src/test/resources/org/opensearch/ml/engine/algorithms/text_similarity/TinyBERT-CE-torch_script.zip?raw=true"
-}
```
Comment (HenryL27) [inline _ml-commons-plugin/custom-local-models.md:349]: Is the `TEXT_SIMILARITY` function name documented anywhere else? If not, we should find a place

File: _ml-commons-plugin/pretrained-models.md (hunk 1 above)
Comment (kolchfa-aws) [reply _ml-commons-plugin/pretrained-models.md:74]: No vector dimension or auto-truncation columns needed per @dhrubo-os

File: _ml-commons-plugin/custom-local-models.md (hunk 2 above)
Comment (kolchfa-aws) [reply _ml-commons-plugin/custom-local-models.md:349]: Makes sense. I added the types to the register model API and to the custom local models page.

**Review by dhrubo-os — APPROVED** [review]

File: _ml-commons-plugin/custom-local-models.md [hunk 3]
```diff
@@ -157,7 +157,7 @@ POST /_plugins/_ml/models/_register
 ```
 {% include copy.html %}
 
-For a description of Register API parameters, see [Register a model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/).
+For a description of Register API parameters, see [Register a model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/). The `model_task_type` corresponds to the model type. For text embedding models, set this parameter to `TEXT_EMBEDDING`. For sparse encoding models, set this parameter to `SPARSE_ENCODING` or `SPARSE_TOKENIZE`. For cross-encoder models, set this parameter to `TEXT_SIMILARITY`.
```
Comment (natebower) [inline _ml-commons-plugin/custom-local-models.md:160]:
```suggestion
For descriptions of Register API parameters, see [Register a model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/register-model/). The `model_task_type` corresponds to the model type. For text embedding models, set this parameter to `TEXT_EMBEDDING`. For sparse encoding models, set this parameter to `SPARSE_ENCODING` or `SPARSE_TOKENIZE`. For cross-encoder models, set this parameter to `TEXT_SIMILARITY`.
```

**Review by natebower — APPROVED** [review]
@kolchfa-aws Just a minor change. Otherwise, LGTM.

**Pull request merged**